### PR TITLE
Add removal endpoints

### PR DESCRIPTION
**Pull request by mikemrm**
This adds removal endpoints for Relationships, Roles and Assignments

### CODE REVIEW

File: internal/api/assignments.go [hunk 1]
```diff
@@ -91,3 +91,47 @@ func (r *Router) assignmentsList(c echo.Context) error {
...
+	}
+
+	subjID, err := gidx.Parse(reqBody.SubjectID)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "error parsing subject ID").SetInternal(err)
+	}
+
+	subjResource, err := r.engine.NewResourceFromID(subjID)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "error deleting assignment").SetInternal(err)
```
Comment (fishnix) [inline internal/api/assignments.go:120]:
```suggestion
		return echo.NewHTTPError(http.StatusBadRequest, "error parsing resource type from subject").SetInternal(err)
```

File: internal/api/relationships.go [hunk 2]
```diff
@@ -124,3 +124,54 @@ func (r *Router) relationshipsList(c echo.Context) error {
...
+
+	ctx, span := tracer.Start(c.Request().Context(), "api.relationshipsDelete", trace.WithAttributes(attribute.String("id", resourceIDStr)))
+	defer span.End()
+
+	resourceID, err := gidx.Parse(resourceIDStr)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "error parsing resource ID").SetInternal(err)
+	}
+
+	var reqBody deleteRelationshipRequest
```
Comment (fishnix) [inline internal/api/relationships.go:139]: Why not use query params for this instead of a DELETE body?

**Review by fishnix — DISMISSED** [review]
just some minor stuff, otherwise looks good 👍

File: internal/api/relationships.go (hunk 2 above)
Comment (mikemrm) [reply internal/api/relationships.go:139]: using a query parameter seems odd for a delete request. however having everything in the url path is pretty standard. I could change it to be `DELETE /resources/:id/relationships/:relation/:related_id`. Thoughts about that?

Comment (fishnix) [reply internal/api/relationships.go:139]: i think this is fine for now 👍

**Review by fishnix — APPROVED** [review]

**Pull request merged**